core/src/test/java/org/elasticsearch/indices/analyze/AnalyzeActionIT.java

```diff
@@ -503,4 +503,20 @@ public void testDetailAnalyzeWithMultiValuesWithCustomAnalyzer() throws Exceptio
 
     }
 
+    public void testNonExistTokenizer() {
+        assertAcked(prepareCreate("test").addAlias(new Alias("alias")));
+        ensureGreen();
+
+        try {
+            AnalyzeResponse analyzeResponse = client().admin().indices()
+                .prepareAnalyze("this is a test")
+                .setAnalyzer("not_exist_analyzer")
+                .get();
+            fail("shouldn't get here");
+        } catch (Throwable t) {
+            assertThat(t, instanceOf(IllegalArgumentException.class));
```

Comment:
Yeah, you are right. Things like IllegalArgumentException should always check it the message. There are exceptions for which its not as important because they are more specific. Though most of the time its just worth checking anyway.